components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -1573,6 +1593,25 @@ interface JsonRpcService {
                             string token_mint_address, string chain_id) =>
       (string amount, uint8 decimals, string uiAmountString,
        SolanaProviderError error, string error_message);
+
+  // Returns the metadata json associated with the NFT. Uses the SimpleHash API, not
+  // the JSON RPC API.
+  GetNftMetadatas(CoinType coin,
+                   array<string> chain_ids,
+                   array<string> contract_addresses,
+                   array<string> token_ids) => (array<NftMetadata> metadatas,
+                                                SolanaProviderError error,
+                                                string error_message);
+
+  // Returns the balances of the list of NFTs for the owner using the SimpleHash API.
+  GetNftBalances(string wallet_address,
+                 array<string> chain_ids,
+                 array<string> contract_addresses,
+                 array<string> token_ids,
+                 CoinType coin) => (array<uint64> balances,
+                                    SolanaProviderError error,
+                                    string error_message);
+
```

Comment:
no need in SolanaProviderError as this should be only for dApps related calls. Just `string error_message` is enough